code/game/gamemodes/cult/blood_magic.dm

```diff
@@ -0,0 +1,816 @@
+				else
+					channeling = FALSE
+					candidate.color = prev_color
+					return
+			else
+				uses--
+				to_chat(user, "<span class='warning'>A dark cloud emanates from you hand and swirls around [candidate] - twisting it into a construct shell!</span>")
+				new /obj/structure/constructshell(T)
+				SEND_SOUND(user, sound('sound/magic/cult_spell.ogg', 0, 1, 25))
+				qdel(candidate)
+		else if(istype(target,/obj/machinery/door/airlock))
+			channeling = TRUE
+			playsound(T, 'sound/machines/airlockforced.ogg', 50, TRUE)
+			do_sparks(5, TRUE, target)
+			if(do_after(user, 50, target = user))
+				if(QDELETED(target))
+					channeling = FALSE
+					return
+				target.narsie_act(TRUE)
+				uses--
+				user.visible_message("<span class='warning'>Black ribbons suddenly emanate from [user]'s hand and cling to the airlock - twisting and corrupting it!</span>")
+				SEND_SOUND(user, sound('sound/magic/cult_spell.ogg', 0, 1, 25))
+				channeling = FALSE
+			else
+				channeling = FALSE
+				return
+		else
+			to_chat(user, "<span class='warning'>The spell will not work on [target]!</span>")
+			return
+		..()
+
+//Armor: Gives the target a basic cultist combat loadout
+/obj/item/melee/blood_magic/armor
+	name = "Arming Aura"
+	desc = "Will equipt cult combat gear onto a cultist on contact."
+	color = "#33cc33" // green
+
+/obj/item/melee/blood_magic/armor/afterattack(atom/target, mob/living/carbon/user, proximity)
+	if(iscarbon(target) && proximity)
+		uses--
+		var/mob/living/carbon/C = target
+		C.visible_message("<span class='warning'>Otherworldly armor suddenly appears on [C]!</span>")
+		C.equip_to_slot_or_del(new /obj/item/clothing/under/color/black(user), slot_w_uniform)
+		C.equip_to_slot_or_del(new /obj/item/clothing/suit/hooded/cultrobes/alt(user), slot_wear_suit)
+		C.equip_to_slot_or_del(new /obj/item/storage/backpack/cultpack(user), slot_back)
+		C.equip_to_slot_or_del(new /obj/item/clothing/shoes/cult(user), slot_shoes)
+		if(C == user)
+			qdel(src) //Clears the hands
+		C.put_in_hands(new /obj/item/melee/cultblade(user))
+		C.put_in_hands(new /obj/item/restraints/legcuffs/bola/cult(user))
+		..()
+
+/obj/item/melee/blood_magic/manipulator
+	name = "Blood Rite Aura"
+	desc = "Absorbs blood from anything you touch. Touching cultists and constructs can heal them. Use in-hand to cast an advanced rite."
+	color = "#7D1717"
+
+/obj/item/melee/blood_magic/manipulator/examine(mob/user)
+	. = ..()
+	. += "Blood spear and blood barrage cost [BLOOD_SPEAR_COST] and [BLOOD_BARRAGE_COST] charges respectively."
+
+/obj/item/melee/blood_magic/manipulator/afterattack(atom/target, mob/living/carbon/human/user, proximity)
+	if(proximity)
+		if(ishuman(target))
+			var/mob/living/carbon/human/H = target
+			if(iscultist(H))
+				if(H.stat == DEAD)
+					to_chat(user,"<span class='warning'>Only a revive rune can bring back the dead!</span>")
+					return
+				if(H.dna && (NO_BLOOD in H.dna.species.species_traits || H.dna.species.exotic_blood != null))
```

Comment:
should it be !no_bood? or am i reading this in reverse again?

Reply:
yeah was wrapping that up, fixed